classes/Settings.js

```diff
@@ -16,66 +64,100 @@ class Settings {
-
-    const configReadableContent = yaml.safeLoad(Base64.decode(configContent));
-    const footerReadableContent = yaml.safeLoad(Base64.decode(footerContent));
+    const navigationContent = fileContentsArr[2].content
 
     // retrieve only the relevant config and index fields
     const configFieldsRequired = {
-      url: configReadableContent.url,
-      title: configReadableContent.title,
-      favicon: configReadableContent.favicon,
-      resources_name: configReadableContent.resources_name,
-      colors: configReadableContent.colors,
+      url: configContent.url,
+      title: configContent.title,
+      favicon: configContent.favicon,
+      shareicon: configContent.shareicon,
+      is_government: configContent.is_government,
+      facebook_pixel: configContent['facebook-pixel'],
+      google_analytics: configContent.google_analytics,
+      resources_name: configContent.resources_name,
+      colors: configContent.colors,
     }
 
     // retrieve footer sha since we are sending the footer object wholesale
     const footerSha = fileContentsArr[1].sha
 
-    return ({ configFieldsRequired, footerContent: footerReadableContent, footerSha })
+    return ({
+      configFieldsRequired,
+      footerContent,
+      navigationContent: { logo: navigationContent.logo },
+      footerSha,
+    })
   }
   
   async post(payload) {
-    // setup 
-    const configResp = new Config(this.accessToken, this.siteName)
-    const config = await configResp.read()
-    const IsomerDataFile = new File(this.accessToken, this.siteName)
-    const dataType = new DataType()
-    IsomerDataFile.setFileType(dataType)
+    const {
+      configResp,
+      FooterFile,
+      NavigationFile,
+      HomepageFile,
+      fileContentsArr,
+    } = await retrieveSettingsFiles(this.accessToken, this.siteName, true)
 
     // extract data
     const {
       footerSettings,
       configSettings,
-      footerSha,
+      navigationSettings,
     } = payload
 
-    // update config object
-    const configContent = yaml.safeLoad(Base64.decode(config.content));
-    Object.keys(configSettings).forEach((setting) => (configContent[setting] = configSettings[setting]));
+    // update settings objects
+    const configContent = fileContentsArr[0].content
+    const footerContent = fileContentsArr[1].content
+    const navigationContent = fileContentsArr[2].content
+
+    const settingsObjArr = [
+      {
+        payload: configSettings,
+        retrievedData: configContent,
+      },
+      {
+        payload: footerSettings,
+        retrievedData: footerContent,
+      },
+      {
+        payload: navigationSettings,
+        retrievedData: navigationContent,
+      },
+    ]
+
+    const updatedSettingsObjArr = settingsObjArr.map(({ payload, retrievedData}) => {
+      const settingsObj = _.cloneDeep(payload);
+      Object.keys(retrievedData).forEach((setting) => (settingsObj[setting] = payload[setting]));
+      return settingsObj
```

Comment:
This portion seems incorrect - we should be cloning the original data (retrievedData) first, then updating it with the key value pairs from the new data (payload)?

So this function should be `const settingsObj = _.cloneDeep(retrievedData);
      Object.keys(payload).forEach((setting) => (settingsObj[setting] = payload[setting]));`

This also allows missing fields in the original data to be filled in

Reply:
You're right, I got confused with the naming of `retrievedData` - resolved with [f5f0d01](https://github.com/isomerpages/isomercms-backend/pull/85/commits/f5f0d01ee925d7e32001fe5a00361f6bda65e64a)